crates/lib/src/qpu/result_data.rs

```diff
@@ -23,7 +24,7 @@ pub enum ReadoutValues {
 }
 
 /// A row of data containing the contents of each memory region at the end of a job.
-#[derive(Debug, Clone, EnumAsInner, PartialEq)]
+#[derive(Debug, Clone, EnumAsInner, PartialEq, Deserialize, Serialize)]
 pub enum MemoryValues {
```

Comment:
for all serialized enums, I might consider `#[serde(tag = "type", content = "content")]` which is easier to deal with in both JS and `jq`. That's stylistic though, happy to hear other opinions.

Reply:
Good point to tag everything consistently. I do think adjacent tags are nice, but we would have to break serialization of RegisterData, as that already uses `#[serde(untagged)]`.

Reply:
Per our chat today, I didn't realize that my request implied a breaking change, I meant to _avoid_ future breaking changes. So I'd suggest one of the following:

* Making these changes in this particular PR (since nothing relies on this yet), and accept inconsistency for the time being
* Being "consistently suboptimal" in enum tagging and following the existing convention

In both cases, this should be registered on the v1.0 wish list for later adoption.